x-pack/plugins/canvas/public/lib/aeroelastic/store.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  ActionId,
+  ChangeCallbackFunction,
+  Meta,
+  NodeResult,
+  Payload,
+  TypeName,
+  UpdaterFunction,
+} from './types';
+
+const makeUid = (): ActionId => 1e11 + Math.floor((1e12 - 1e11) * Math.random());
```

Comment:
This generates a random ID, but not a unique ID, so there could be collisions. Since it's used in `select` for caching, it seems pretty important that you don't have collisions. Why not just use the `uuid` library here? Or even just keep incrementing a number?

Reply:
Good point, updated to a plain counter